kiwi/builder/install.py

```diff
@@ -281,6 +281,8 @@ def create_install_pxe_archive(self) -> None:
             keep_source_on_compress=True
         )
         compress.xz(self.xz_options)
+        # set by compress.xz
+        assert compress.compressed_filename
```

Comment:
Is this assert useful ?

* if compress.xz() fails it will raise an Exception from the Command.run
* I don't see how the AssertionError could be reached
* if it is reached all we will get is "AssertionError" without any further information

compress.xz() also returns ```compressed_filename``` I think it might be better to refactor the code in another pull request that to use the return value from ```compress.xz()``` rather than accessing the member of the compress object after the call.

In general I'm not a friend of assertions, they are usually placed at code that should never be reached but "who knows" :) if possible I think such code elements should be avoided. Thoughts ?

Reply:
You are right, I only added this to work around the type checker, as it would otherwise not accept my code. I have pushed another commit that removes the usage of `Compress.compressed_filename` everywhere and hence removes this problem.